### ISSUE TITLE
Dynamic Key Question

### ISSUE THREAD

**Issue by tqphan** [OptiKey/OptiKey#919]
How do I create a dynamic key that only write to the scratch pad and does not actually simulate the key to the active window? I need to create a key that behaves similar to a key in the Conversation Keyboard. Is it possible to create a key that behaves that way?

**Comment by kmcnaught** [comment]
I don't believe that is currently possible. At the moment, when a `Keyboard` class is created (be it a dynamic keyboard, or a built-in Optikey one) the class gets a `SimulateKeyStrokes` bool which is passed to the `KeyboardOutputService`. A built-in conversation keyboard is set up to suppress simulated key strokes (`SimulateKeyStrokes=false`) on all keys. A dynamic keyboard isn't. 

What's the context? Depending on what you are trying to achieve, we could consider:
- Adding a global `SuppressSimulatedKeyStrokes` setting (this would affect all keyboards but be trivial to implement)
- Adding a `SimulateKeyStrokes` boolean as a keyboard-level setting in an XML keyboard (this would affect all keys on the keyboard)
- Adding a `SuppressSimulatedKeyStrokes` property to an individual dynamic key

**Comment by tqphan** [comment]
Thanks for the reply. I'm trying to create a custom Conversion Keyboard. Option 2 or 3 would be the most useful for my case.

**Comment by bigmutty** [comment]
> Thanks for the reply. I'm trying to create a custom Conversion Keyboard. Option 2 or 3 would be the most useful for my case.

I second that.

**Comment by kmcnaught** [comment]
I have an implementation for option (2) ready for my next PR.
